AST UnaryOperation node with `cannot overflow` string after operator

The `UnaryOperation` node may have a `cannot overflow` string after the operator in Clang 9.0.

# Example

Contents of `e.c`:
```c
int f(int x) {
	if (!x) {
		return 42;
	}
	return 10;
}
```

# Panic

```
$ c2go transpile e.c
panic: could not match regexp with string
^(?P<address>[0-9a-fx]+) <(?P<position>.*)> '(?P<type>.*?)'(:'(?P<type2>.*)')?(?P<lvalue> lvalue)?(?P<prefix> prefix)?(?P<postfix> postfix)? '(?P<operator>.*?)'[\s]*$
0x5654c18208f8 <line:2:6, col:7> 'int' prefix '!' cannot overflow


goroutine 39 [running]:
github.com/elliotchance/c2go/ast.groupsFromRegex(0xc00030e000, 0xa6, 0xc000160539, 0x41, 0xc000032700)
```

# AST output

The AST of `e.c` looks as follows:

```
$ c2go ast e.c
TranslationUnitDecl 0x5615bc7209f8 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x5615bc721290 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x5615bc720f90 '__int128'
|-TypedefDecl 0x5615bc721300 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x5615bc720fb0 'unsigned __int128'
|-TypedefDecl 0x5615bc7215e8 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x5615bc7213e0 'struct __NSConstantString_tag'
|   `-Record 0x5615bc721358 '__NSConstantString_tag'
|-TypedefDecl 0x5615bc721680 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x5615bc721640 'char *'
|   `-BuiltinType 0x5615bc720a90 'char'
|-TypedefDecl 0x5615bc721958 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x5615bc721900 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x5615bc721760 'struct __va_list_tag'
|     `-Record 0x5615bc7216d8 '__va_list_tag'
`-FunctionDecl 0x5615bc77f7d0 </home/u/Desktop/ccc/e/e.c:1:1, line:6:1> line:1:5 f 'int (int)'
  |-ParmVarDecl 0x5615bc77f700 <col:7, col:11> col:11 used x 'int'
  `-CompoundStmt 0x5615bc77f9a0 <col:14, line:6:1>
    |-IfStmt 0x5615bc77f958 <line:2:2, line:4:2>
    | |-UnaryOperator 0x5615bc77f8f8 <line:2:6, col:7> 'int' prefix '!' cannot overflow
    | | `-ImplicitCastExpr 0x5615bc77f8e0 <col:7> 'int' <LValueToRValue>
    | |   `-DeclRefExpr 0x5615bc77f8c0 <col:7> 'int' lvalue ParmVar 0x5615bc77f700 'x' 'int'
    | `-CompoundStmt 0x5615bc77f940 <col:10, line:4:2>
    |   `-ReturnStmt 0x5615bc77f930 <line:3:3, col:10>
    |     `-IntegerLiteral 0x5615bc77f910 <col:10> 'int' 42
    `-ReturnStmt 0x5615bc77f990 <line:5:2, col:9>
      `-IntegerLiteral 0x5615bc77f970 <col:9> 'int' 10
```

--- a/ast/unary_operator.go
+++ b/ast/unary_operator.go
@@ -2,14 +2,15 @@ package ast
 
 // UnaryOperator is type of unary operator
 type UnaryOperator struct {
-	Addr       Address
-	Pos        Position
-	Type       string
-	Type2      string
-	IsLvalue   bool
-	IsPrefix   bool
-	Operator   string
-	ChildNodes []Node
+	Addr           Address
+	Pos            Position
+	Type           string
+	Type2          string
+	IsLvalue       bool
+	IsPrefix       bool
+	Operator       string
+	CannotOverflow bool
+	ChildNodes     []Node
 }
 
 func parseUnaryOperator(line string) *UnaryOperator {
@@ -19,19 +20,21 @@ func parseUnaryOperator(line string) *UnaryOperator {
 		(?P<lvalue> lvalue)?
 		(?P<prefix> prefix)?
 		(?P<postfix> postfix)?
-		 '(?P<operator>.*?)'`,
+		 '(?P<operator>.*?)'
+		(?P<cannot_overflow> cannot overflow)?`,
 		line,
 	)
 
 	return &UnaryOperator{
-		Addr:       ParseAddress(groups["address"]),
-		Pos:        NewPositionFromString(groups["position"]),
-		Type:       groups["type"],
-		Type2:      groups["type2"],
-		IsLvalue:   len(groups["lvalue"]) > 0,
-		IsPrefix:   len(groups["prefix"]) > 0,
-		Operator:   groups["operator"],
-		ChildNodes: []Node{},
+		Addr:           ParseAddress(groups["address"]),
+		Pos:            NewPositionFromString(groups["position"]),
+		Type:           groups["type"],
+		Type2:          groups["type2"],
+		IsLvalue:       len(groups["lvalue"]) > 0,
+		IsPrefix:       len(groups["prefix"]) > 0,
+		Operator:       groups["operator"],
+		CannotOverflow: len(groups["cannot_overflow"]) > 0,
+		ChildNodes:     []Node{},
 	}
 }
 

--- a/ast/unary_operator_test.go
+++ b/ast/unary_operator_test.go
@@ -46,6 +46,17 @@ func TestUnaryOperator(t *testing.T) {
 			Operator:   "*",
 			ChildNodes: []Node{},
 		},
+		`0x55a7effc0878 <line:69:13, col:59> 'int' prefix '!' cannot overflow`: &UnaryOperator{
+			Addr:           0x55a7effc0878,
+			Pos:            NewPositionFromString("line:69:13, col:59"),
+			Type:           "int",
+			Type2:          "",
+			IsLvalue:       false,
+			IsPrefix:       true,
+			Operator:       "!",
+			CannotOverflow: true,
+			ChildNodes:     []Node{},
+		},
 	}
 
 	runNodeTests(t, nodes)

